hugolib/site.go

```diff
@@ -1431,27 +1432,47 @@ func (s *Site) assembleMenus() {
 	}
 
 	sectionPagesMenu := s.Info.sectionPagesMenu
-	sectionPagesMenus := make(map[string]interface{})
-	//creating flat hash
 	pages := s.Pages
-	for _, p := range pages {
 
-		if sectionPagesMenu != "" {
+	if sectionPagesMenu != "" {
+		// Create menu entries for section pages with content
+		for _, p := range pages {
+			if p.Section() != "" && p.Kind == KindSection {
+				me := MenuEntry{Identifier: p.Section(),
+					Name:   p.LinkTitle(),
+					Weight: p.Weight,
+					URL:    p.RelPermalink()}
+
+				// menu with same id defined in config, let that one win
+				if _, ok := flat[twoD{sectionPagesMenu, me.KeyName()}]; ok {
+					continue
+				}
+				flat[twoD{sectionPagesMenu, me.KeyName()}] = &me
+			}
+		}
+
+		// Create entries for remaining content-less section pages
```

Comment:
To be honest... I forgot most of it.

This piece of code was kind of ugly. It ran through all pages, collecting a list of all sections they belong to, and from this collection of section, synthesized menu entries.

What I remember, is that I thought I could do better, by running through all section pages, and assembling the menu entry via some method in Page, using the actual page metadata. It didn't work, it turns out there was a reason for doing the old ugly thing. Probably content-less section pages are not yet there when the menu is assembled.

Then I ended up refining the old method, by using the section page if it is there, and doing the old thing of synthesizing a menu entry for mentioned sections (not actual section pages) otherwise.

Maybe you, that are more familiar with the code can fill in the blanks. Otherwise I can try squeeze my brain a little more and see if anything else comes out.